src/twisted/internet/_newtls.py

```diff
@@ -159,7 +159,13 @@ def startTLS(transport, contextFactory, normal, bypass):
 
 
 @implementer(ITLSTransport)
-class ConnectionMixin(object):
+class ConnectionMixin(object):  # type: ignore[misc]
```

Comment:
I see that this is only used in a single place

src/twisted/internet/tcp.py
```
@implementer(interfaces.ITCPTransport, interfaces.ISystemHandle)
class Connection(_TLSConnectionMixin, abstract.FileDescriptor, _SocketCloser,
                 _AbortingMixin):
```

So maybe this should not be a mixin, but rather a full transport

```suggestion
class ConnectionMixin(abstract.FileDescriptor):
```

but I see that abstract.FileDescriptor is not really fully implementing the ITransport interface

I guess is best to leave this as a separate ticket.

So instead of `# XXX` create a ticket and add the link to that ticket in the comment to mark is as a "TODO" or "FIXME"

Reply:
Since `ConnectionMixin` is defined in `_newtls.py`, maybe we can assume that entire module is private?
If so, would it be better to remove @implementer(ITLSTransport) from the ConnectionMixin, and add it to `twisted.internet.tcp.Connection`?

I really don't like how some of these interfaces have been defined and inherit from each other....

Reply:
I've created a new PR for this: https://github.com/twisted/twisted/pull/1299

and removed the changes to ConnectionMixin from here.